test/absinthe/resolution_test.exs

```diff
@@ -66,6 +101,54 @@ defmodule Absinthe.ResolutionTest do
     assert ["id", "name"] == fields
   end
 
+  test "project/1 works with interfaces" do
+    doc = """
+    {
+      named {
+        ... on User {
+          id
+        }
+
+        name
+
+        ... on Member {
+          mid
+        }
+      }
+    }
+    """
+
+    {:ok, _} = Absinthe.run(doc, Schema)
+
+    assert_receive({:fields, fields})
+
+    assert ["id", "name", "mid"] == fields
```

Comment:
An abstract field resolves to exactly _one_ concrete type for any given resolver invocation. There is no concrete type where `["id", "name", "mid"]` are all valid fields.

Reply:
The type of `named` query is `named` that is an interface, and `user` and `member` implement it, that’s why returning with `["id", "name", "mid"]` that isn’t a concrete type.

Reply:
The point of project however is to return the concrete fields to execute, not all fields that may be possibly executed by some type. `project/1` should honestly raise when provided an abstract type from the parent and no concrete type name to project on to.

Reply:
The function you want to use is this: https://hexdocs.pm/absinthe/Absinthe.Resolution.html#project/2

`Resolution.project(info, :user)` or `Resolution.project(info, :member)`. Then you'll get the fields for each of those types. That's what I mean about `Resolution.project(info)` basically being invalid for abstract types.

Reply:
Thanks, I got it